File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/domain/AuditableLongEntity.java
package com.coupang.springframework.data.requery.domain;

import io.requery.*;
import io.requery.converter.LocalDateTimeConverter;

import java.time.LocalDateTime;

/**
 * com.coupang.springframework.data.requery.domain.AuditableLongEntity
 *
 * @author debop
 * @since 18. 6. 4
 */
@Superclass
public abstract class AuditableLongEntity extends AbstractPersistable<Long> {

    @Convert(LocalDateTimeConverter.class)
    protected LocalDateTime createdDate;

    @Lazy
    @Convert(LocalDateTimeConverter.class)
    protected LocalDateTime lastModifiedDate;


    @PreInsert
    protected void onPreInsert() {
        createdDate = LocalDateTime.now();
    }

    @PreUpdate
    protected void onPreUpdate() {
        lastModifiedDate = LocalDateTime.now();
    }

    private static final long serialVersionUID = -6288516556005303756L;
}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/domain/AuditableLongEntity.kt
package org.springframework.data.requery.domain

import io.requery.*
import org.springframework.data.requery.converters.LocalDateTimeToLongConverter
import java.time.LocalDateTime

/**
 * org.springframework.data.requery.domain.AuditableLongEntity
 *
 * @author debop
 */
@Superclass
abstract class AuditableLongEntity: AbstractPersistable<Long>() {

    @get:Lazy
    @get:Convert(LocalDateTimeToLongConverter::class)
    abstract var createdAt: LocalDateTime

    @get:Lazy
    @get:Convert(LocalDateTimeToLongConverter::class)
    abstract var modifiedAt: LocalDateTime

    @PreInsert
    protected fun onPreInsert() {
        createdAt = LocalDateTime.now()
    }

    @PreUpdate
    protected fun onPreUpdate() {
        modifiedAt = LocalDateTime.now()
    }

}